keel-core/src/main/kotlin/com/netflix/spinnaker/keel/annotation/ForcesNew.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.keel.annotation
+
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaInject
+
+/**
+ * Flags IntentSpec parameters that cannot change once created without
+ * recreating the intent.
+ */
+@JsonSchemaInject(json = "{ \"forcesNew\": true }")
+@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ForcesNew(val value: Boolean = true)
```

Comment:
Why don't you call this `Immutable`?

Reply:
I originally had called it this, but I wanted to communicate the side effects of changing the value. It's also consistent with Terraform's taxonomy, which I thought may help some people onboard. Do you have strong feelings here?

Reply:
Oh so you're communicating that if the value changes, it will make a new object? That's fine then. I thought `Immutable` might be clearer but it sounds like it won't be.